### PR TITLE
fix: zoom shows up as location when expired instead of cal

### LINKED ISSUE #14858: [CAL-3619] Zoom is using Cal video instead (likely expired)
It seems like when Zoom expires, it falls back to Cal Video but it still says Zoom everywhere.

What is the status of our "Zoom credentials expired. Please reinstall" Alert Emails?

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3619](https://linear.app/calcom/issue/CAL-3619/zoom-is-using-cal-video-instead-likely-expired)</sub>

### PR DESCRIPTION
**Pull request by Shaik-Sirajuddin**
## What does this PR do?

- Fixes #14858 
- Fixes CAL-3619

Before : [Screenshot](https://github.com/calcom/cal.com/issues/14858#issue-2275433877)
After : [Loom Video](https://www.loom.com/share/7701cb3edcbe48e79f94801fd3b06de3?sid=2bdfb484-4515-4d3c-a4b9-ece1815031ef)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?
-Install zoom or any other conferencing app
-Edit an event to set location to the installed app
-Modify respective apps credential in db , to make credential invalid 
-Book an event with an alternative account
-location will as cal video ( with calvideo link ) , as video meeting url creation failed with freshly installed app
-this was not the case before the fix

### CODE REVIEW

File: packages/core/videoClient.ts
Comment (Udit-takkar) [inline packages/core/videoClient.ts:121]: use variable `DailyLocationType` from packages/app-store/locations.ts

File: packages/core/videoClient.ts
Comment (Udit-takkar) [reply packages/core/videoClient.ts:124]: IIRC originalEvent is only reserved for storing the original event before rescheduling

Comment (Shaik-Sirajuddin) [reply packages/core/videoClient.ts:124]: originalEvent's location property is being used to update the actual event in EventManager's create function
the location is later used to update booking location while creating booking references

Comment (Shaik-Sirajuddin) [reply packages/core/videoClient.ts:124]: originalEvent was modified while defaulting to calVideo , but the returnObject consiting of originalEvent is not modified.
this line modifies the returnObject to update originalEvent  property

` returnObject = { ...returnObject, originalEvent: calEvent, createdEvent: defaultMeeting };`

Comment (Shaik-Sirajuddin) [reply packages/core/videoClient.ts:125]: other than updating event location , originalEvent from videoClient is not used as part of logic or db updation.

File: packages/core/videoClient.ts
Comment (Shaik-Sirajuddin) [reply packages/core/videoClient.ts]: Updated, PTAL @Udit-takkar

**Review by Udit-takkar — COMMENTED** [review]
@Shaik-Sirajuddin Looks good. could you also add a unit test or e2e test for this?

**Comment by Shaik-Sirajuddin** [comment]
> @Shaik-Sirajuddin Looks good. could you also add a unit test or e2e test for this?

Sure @Udit-takkar

**Comment by Shaik-Sirajuddin** [comment]
A tests already exists to cover the scenario when the zoom meeting creation fails
I have modified the test to ensure location is updated to dailyvideo
PTAL @Udit-takkar

**Review by Udit-takkar — APPROVED** [review]
LGTM. I'll wait for one more review before merging

**Review by sean-brydon — APPROVED** [review]
LGTM also - nice job. 

Thanks for tests!

**Pull request merged**
